packages/x-data-grid-premium/src/components/pivotPanel/GridPivotPanelSearch.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { GridSlotProps, getDataGridUtilityClass } from '@mui/x-data-grid-pro';
+import { styled } from '@mui/system';
+import { vars } from '@mui/x-data-grid-pro/internals';
+import { unstable_composeClasses as composeClasses } from '@mui/utils';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { DataGridPremiumProcessedProps } from '../../models/dataGridPremiumProps';
+
+export type GridPivotPanelSearchProps = GridSlotProps['baseTextField'] & {
+  onClear: () => void;
+};
+
+type OwnerState = DataGridPremiumProcessedProps;
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    container: ['pivotPanelSearchContainer'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const GridPivotPanelSearchContainer = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'PivotPanelSearchContainer',
+})<{ ownerState: OwnerState }>({
+  padding: vars.spacing(0, 1, 1),
+});
+
+function GridPivotPanelSearch(props: GridPivotPanelSearchProps) {
+  const { onClear, ...other } = props;
+  const rootProps = useGridRootProps();
+  const apiRef = useGridApiContext();
+  const classes = useUtilityClasses(rootProps);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === 'Escape') {
+      onClear();
+    }
+
+    props.onKeyDown?.(event);
```

Comment:
Would non-null assertion operator make more sense here since it's been used internally?
```suggestion
    props.onKeyDown!(event);
```

Reply:
`onKeyDown` seems unnecessary here since it's never passed, I'll remove it instead.

Reply:
[`112eb8c` (#9877)](https://github.com/mui/mui-x/pull/9877/commits/112eb8ca5eaebf38df79a98b1061511ab821b461)

@KenanYusuf let me know if you disagree